src/components/gallery/AddList.tsx

```diff
@@ -0,0 +1,90 @@
+import * as style from "./AddListStyle";
+import Input from "../common/Input";
+import Button from "../common/Button";
+import { useState } from "react";
+import { collection, addDoc, getFirestore } from "firebase/firestore";
+import { app } from "@/firebase/firebase";
+
+const firestore = getFirestore(app);
+
+interface Folders {
+  createdAt: Date;
+  id: string;
+  sub: string[];
+  title: string;
+}
+
+interface GallerySideProps {
+  closeAddListModal: () => void;
+  galleryData: Folders[];
+  setGalleryData: React.Dispatch<React.SetStateAction<Folders[]>>;
+}
+
+export default function Addlist({
+  closeAddListModal,
+  galleryData,
+  setGalleryData,
+}: GallerySideProps) {
+  const [inputValue, setInputValue] = useState("");
+
+  const createGallery = async () => {
+    try {
+      const timestamp = new Date().getTime();
+      const randomValue = Math.random();
+      const uniqueId = `${timestamp}-${randomValue}`;
+
+      const newData = {
+        title: inputValue,
+        createdAt: new Date(),
+        sub: [],
+        id: uniqueId,
+      };
+
+      const docRef = await addDoc(collection(firestore, "Gallery"), newData);
+
+      const copy = [...galleryData];
+      copy.push(newData);
+      setGalleryData(copy);
+      closeAddListModal();
+      console.log("새로운 Gallery 문서가 생성되었습니다. 문서 ID:", docRef.id);
+    } catch (error) {
+      console.error("Gallery 생성 중 오류 발생:", error);
+    }
+  };
+
+  const setValue = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setInputValue(e.target.value);
+  };
+
+  const saveTitle = () => {
+    console.log(inputValue);
+    createGallery();
+  };
+
+  return (
+    <style.Container>
+      <style.Form>
+        <div>
+          <label htmlFor="title" style={{ marginRight: "1rem" }}>
+            이름
+          </label>
+          <Input type="text" id="title" onChange={(e) => setValue(e)} />
```

Comment:
```suggestion
          <Input type="text" id="title" onChange={setValue} />
```

이렇게 해도 될까요, 안 될까요??

된다면 왜 위 방식이 더 좋은지와 안 된다면 왜 안 되는지를 확인해보세요!